pkg/codegen/schema/loader.go

```diff
@@ -38,7 +43,7 @@ func (l *pluginLoader) getPackage(key string) (*Package, bool) {
 }
 
 // ensurePlugin downloads and installs the specified plugin if it does not already exist.
-func (l *pluginLoader) ensurePlugin(pkg string, version *semver.Version) error {
+func ensurePlugin(pkg string, version *semver.Version) error {
```

Comment:
Did you move this off of `pluginLoader` b/c it doesn't actually need any of the loader's fields?

Reply:
Yeah, to make it easier to test in my hacky test which I'm not checking in. I can revert.